plugins/rbac-backend/src/service/role-manager.ts

```diff
@@ -142,6 +227,14 @@ export class BackstageRoleManager implements RoleManager {
     if (!memo.isAcyclic()) {
       const cycles = memo.findCycles();
 
+      if (!memo.hasEntityRef(name2) && this.parseEntity(name2) === 'role') {
```

Comment:
I guess that's a wrong place for this code block. `if (!memo.isAcyclic()) {` that's code block which should notify policy admin about cycle dependency - mistake in the group structure.

When you provide return with result, then you hide warning.

```
if (this.parseEntity(name2) === 'role') {
  for (const [key, value] of this.allRoles.entries()) {
    return (
        value.hasRole(tempRoleName2, 1) &&
         memo.hasEntityRef(this.parseRole(key))
    );
  }  
}
```

Reply:
So would removing the return statement fix the issue? Something like this:
```
if (!memo.hasEntityRef(name2) && this.parseEntityKind(name2) === 'role') {
  for (const [key, value] of this.allRoles.entries()) {
    value.hasRole(tempRoleName2, 1);
    memo.hasEntityRef(key);
  }
}
```

Reply:
I think in this case code do nothing, so this code can be removed. I guess in case cycle dependency it is better to return always false for has link with warning message in the log. Also you can modify error message to notify user, that role permission was also affected with some group which contain cycle dependency.